File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        d={}
        start=0
        ans=0
        for i in range(len(s)):
            if s[i] in d and d[s[i]]>=start:
                start=d[s[i]]+1
            d[s[i]]=i
            ans=max(ans,i-start+1)
        return ans

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character,Integer> m=new HashMap<>();
        int start=0;
        int ans=0;
        for(int i=0;i<s.length();i++){
            char c=s.charAt(i);
            if(m.containsKey(c) && m.get(c)>=start){
                start=m.get(c)+1;
            }
            ans=Math.max(ans,i-start+1);
            m.put(c,i);
        }
        return ans;
    }
}